q2_cutadapt/_demux.py

```diff
@@ -276,21 +285,37 @@ def demux_paired(seqs: MultiplexedPairedEndBarcodeInSequenceDirFmt,
     _check_barcodes_uniqueness(
         forward_barcodes, reverse_barcodes, mixed_orientation)
 
+    if (
+        not mixed_orientation
+        and anchor_reverse_barcode and (reverse_barcodes is None)
+    ):
+        raise ValueError("A reverse barcode needs to be provided in order to "
+                         "anchor the reverse barcode.")
+
     if (
         mixed_orientation
         and forward_cut != reverse_cut
     ):
         raise ValueError("'forward_cut' and 'reverse_cut' need to be set to "
                          "the same number when using the 'mixed_orientation' "
-                         "mode")
+                         "mode.")
+
+    if (
+        mixed_orientation
+        and anchor_forward_barcode != anchor_reverse_barcode
+    ):
+        raise ValueError(
+            "'anchor_forward_barcode' and 'anchor_reverse_barcode' need to be "
+            "set to the same value when using the 'mixed_orientation' mode."
+        )
```

Comment:
This is a little captious but could you put these checks in a helper? You can just pass the `locals()` from this function to that helper so the signature isn't ridiculous.